src/specBuilder/legend/legendSpecBuilder.ts

```diff
@@ -278,9 +301,25 @@ export const addData = produce<Data[], [LegendSpecProps & { facets: string[] }]>
 );
 
 export const addSignals = produce<Signal[], [LegendSpecProps]>(
-	(signals, { hiddenSeries, highlight, isToggleable, legendLabels, name }) => {
+	(signals, { hiddenSeries, highlight, isToggleable, legendLabels, name, animations }) => {
+		// if animations are enabled, add all necessary animation signals.
+		//TODO: add tests
+		if (animations) {
+			// we don't want to add the signals if they already exist.
+			// Since they come as a package, we only need to check if one animation signal is present.
```

Comment:
@c-lamoureux since we get all of the animation signals together in getRscAnimationSignals() we should also want to get only the signals that are missing, right? This would need a refactor of getRscAnimationSignals.

Reply:
This has been changed